pkg/network/ebpf/c/protocols/classification/routing.h

```diff
@@ -61,15 +61,16 @@ static __always_inline void init_routing_cache(usm_context_t *usm_ctx, protocol_
     }
 
     // We skip a given layer in two cases:
-    // 1) If the protocol for that layer is known
+    // 1) If the protocol for that layer is known, 
+    //    except for encryption as it still needs to be traversed for metadata
     // 2) If there are no programs registered for that layer
     if (stack->layer_application || !has_available_program(__PROG_APPLICATION)) {
         usm_ctx->routing_skip_layers |= LAYER_APPLICATION_BIT;
     }
     if (stack->layer_api || !has_available_program(__PROG_API)) {
         usm_ctx->routing_skip_layers |= LAYER_API_BIT;
     }
-    if (stack->layer_encryption || !has_available_program(__PROG_ENCRYPTION)) {
+    if (stack->flags == FLAG_FULLY_CLASSIFIED || !has_available_program(__PROG_ENCRYPTION)) {
```

Comment:
why only when fully classified?
I think this is going to create inaccuracies for USM

Reply:
responded [here](https://github.com/DataDog/datadog-agent/pull/31464/files#r1880025663)